test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -79,25 +68,88 @@ var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func(
 				"cluster-scoped-resources/machineconfiguration.openshift.io/kubeletconfigs/performance-performance.yaml",
 				"namespaces/openshift-cluster-node-tuning-operator/tuned.openshift.io/tuneds/openshift-node-performance-performance.yaml",
 			}
-			err := checkfilesExist(ClusterSpecificFiles, mustGatherContentDir)
-			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Checking Folder: %q\n", mgContentFolder))
+			By("\tLooking for generic files")
+			err := checkfilesExist(ClusterSpecificFiles, mgContentFolder)
+			Expect(err).ToNot(gomega.HaveOccurred())
 		})
+
 		It("Verify hardware related information are captured", func() {
+
+			var workerRTNodes []corev1.Node
+
+			workerRTNodes, err := nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+			Expect(err).ToNot(HaveOccurred())
+			cnfWorkerNode := workerRTNodes[0].ObjectMeta.Name
+
+			// find the path of sysinfo.tgz of the correct node
+			snapShotName := ""
+			err = filepath.Walk(mgContentFolder,
+				func(path string, info os.FileInfo, err error) error {
+					if err != nil {
+						return err
+					}
+					if !info.IsDir() && info.Name() == "sysinfo.tgz" {
+						if strings.Contains(path, cnfWorkerNode) {
+							snapShotName = path
+						}
+					}
+					return nil
+				})
+			if err != nil {
+				log.Println(err)
+			}
+
+			// find the path of must-gather node files
```

Comment:
why do we need to do this again?

Reply:
Because ghw unzip the tgz in a tmp folder not in snapShotDir

Reply:
very fair. Please record this as comment in the source code.

Reply:
Add comment. Done

Reply:
perfect, thanks